paramak/parametric_shapes/sweep_mixed_shape.py

```diff
@@ -0,0 +1,159 @@
+from collections import Iterable
+
+import cadquery as cq
+
+from paramak import Shape
+
+
+class SweepMixedShape(Shape):
+    """Sweeps a 2D shape created from points connected with straight, spline or circle
+    connections along a defined spline path to create a 3D CadQuery solid.
+
+    Args:
+        points (list of tuples each containing X (float), Z (float), connection): A list
+            of XY, YZ or XZ coordinates connected by straight connections which define
+            the 2D shape to be swept. The coordinates are defined with respect to origins
+            at the first and last points of the spline path.
+        path_points (list of tuples each containing X (float), Z (float)): A list of XY,
+            YZ or XZ coordinates connected by spline connections which define the path
+            along which the 2D shape is swept.
+        workplane (str): Workplane in which the 2D shape to be swept is defined.
+        path_workplane (str): Workplane in which the spline path is defined.
+    """
```

Comment:
Here some non-default arguments appear to be missing (stp_filename and stl_filename) as well as default values for workplane and path_workplane